docs/api/hooks.md

```diff
@@ -69,19 +71,119 @@ export const CounterComponent = () => {
 }
 ```
 
-Using props to determine what to extract:
+Using props via closure to determine what to extract:
 
 ```jsx
 import React from 'react'
 import { useSelector } from 'react-redux'
 
 export const TodoListItem = props => {
-  const todo = useSelector(state => state.todos[props.id], [props.id])
-
+  const todo = useSelector(state => state.todos[props.id])
```

Comment:
I *think* that perhaps the docs in this case should encourage the usage of `useCallback`?

```js
const todoSelector = useCallback(state => state.todos[props.id], [props.id])
const todo = useSelector(todoSelector)
```

FTR I actually think that this is a good practice regardless of what ends up happening with #1273 :slightly_smiling_face:

Reply:
But why should we encourage usage of `useCallback`? I see no real benefit (the cost of creating new lamba per render is negligible) and it only adds complexity.

Reply:
Yep @lukeapage you are right. Actually, that would be likely to make perf even worse because the overhead of an extra hook. Yep, that was a brain-fart. Sorry!

Reply:
@josepot you meant @lukaszfiszer ?

Reply:
Yep, I did, indeed mean @lukaszfiszer ... Sorry about that! :sweat_smile: